datapath: probe socket match support, plumb to Envoy configuration

jrajahalme opened the pull request:
Let Envoy know if it may use original source addresses in upstream
connections. Limit support to egress proxy only for now.

This fixes policy regressions when the iptables socket match kernel
module is not available by not using original source address in proxy
upstream connections in that case.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/9016)
<!-- Reviewable:end -->

brb commented on pkg/datapath/iptables/iptables.go:
```diff
@@ -334,7 +334,16 @@ func (m *IptablesManager) Init() {
-	m.ip6tables = ip6tables
+	m.haveIp6tables = ip6tables
+
+	if err := modulesManager.FindOrLoadModules("xt_socket"); err != nil {
```
Shouldn't we probe for the `nf_socket_ipv{4,6}` modules as well?

```
> lsmod | grep xt_socket
xt_socket              16384  0
nf_socket_ipv4         16384  1 xt_socket
nf_socket_ipv6         20480  1 xt_socket
nf_defrag_ipv6         24576  2 nf_conntrack,xt_socket
nf_defrag_ipv4         16384  2 nf_conntrack,xt_socket
x_tables               49152  8 xt_conntrack,iptable_filter,xt_socket,xt_tcpudp,xt_addrtype,xt_nat,ip_tables,xt_MASQUERADE
```

joestringer replied:
Given that they are dependencies, I think that if they are not available then `xt_socket` load will fail anyway. Unless you're concerned about kernels with IPv6 compiled out?

brb replied:
Makes sense.

jrajahalme committed:
envoy: Update to the latest API

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

jrajahalme committed:
datapath: probe socket match support, plumb to Envoy configuration

Let Envoy know if it may use original source addresses in upstream
connections. Limit support to egress proxy only for now.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>
